.github/workflows/deploy-image-on-master-merge.yaml

```diff
@@ -0,0 +1,79 @@
+name: Deploy Application images when Merging on Master
+
+on:
+  # run it on push to the default repository branch
+  push:
+    branches: [master]
+  # For testing purpose only
+  pull_request:
+
+env:
+  REGISTRY: ghcr.io
+  REPOSITORY_NAME: ${{ github.repository }}
+
+jobs:
+  # define job to build and publish docker image
+  build-and-push-docker-image:
+    name: Build Docker image and push to repositories
+    # run only when code is compiling and tests are passing
+    runs-on: ubuntu-latest
+    strategy:
+      fail-fast: false
+      matrix: # Define a matrice to build each images and tag it
+        include:
+          - dockerfile: ./listingapi/Dockerfile
+            image: listingapi
+          - dockerfile: ./pricemap-python/Dockerfile
+            image: pricemap-python
+          - dockerfile: ./pricemap-typescript/Dockerfile
+            image: pricemap-typescript
+
+    # steps to perform in job
+    steps:
+      - name: Checkout code
+        uses: actions/checkout@v3
+
+      # Github doesn't allow uppercase on registry name
+      - name: set lower case repository name
+        id: lowercase
+        run: echo "repository_name_lower_case=${REPOSITORY_NAME,,}" >>$GITHUB_OUTPUT
+        env:
+          REPOSITORY_NAME: '${{ env.REPOSITORY_NAME }}'
+
+      # Use Emulator to build images on different architectures
+      - name: Set up QEMU
+        uses: docker/setup-qemu-action@master
+        with:
+          platforms: all
+
+      # Use BuildX to build the docker images
+      - name: Set up Docker Buildx
+        id: buildx
+        uses: docker/setup-buildx-action@v2
+
+      - name: Login to Github Packages
+        uses: docker/login-action@v2
+        with:
+          registry: ${{ env.REGISTRY }}
+          username: ${{ github.actor }}
+          password: ${{ secrets.GITHUB_TOKEN }}
+
+      # Get the current date and time to tag images
+      - name: Get current date and time
+        id: date
+        run: echo "date=$(date +'%Y-%m-%d_%H-%M-%S')" >> $GITHUB_OUTPUT
+
+      # For each image, build it, tag and push it on registry
+      - name: Build image and push to Registry
+        id: docker_build
+        uses: docker/build-push-action@v2
+        with:
+          platforms: linux/amd64, linux/arm/v7, linux/arm64/v8
+          file: ${{ matrix.dockerfile }}
+          push: true
+          tags: |
```

Comment:
We could use https://github.com/docker/metadata-action to generate the tags beforehand in the workflow.